test/e2e/tests/test_table_replicas.py

```diff
@@ -0,0 +1,542 @@
+        for region in regions_to_remove:
+            assert region not in region_names
+
+    def test_delete_table_with_replicas(self, table_with_replicas):
+        (ref, res) = table_with_replicas
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Delete the k8s resource
+        k8s.delete_custom_resource(ref)
+
+        max_wait_seconds = REPLICA_WAIT_AFTER_SECONDS
+        interval_seconds = 30
+        start_time = time.time()
+
+        while time.time() - start_time < max_wait_seconds:
+            if not self.table_exists(table_name):
+                break
+            time.sleep(interval_seconds)
+
+        # Verify the table was deleted
+        assert not self.table_exists(table_name)
+        replicas = table.get_replicas(table_name)
+        assert replicas is None
+
+    def test_terminal_condition_for_invalid_stream_specification(self, table_with_invalid_replicas):
+        (ref, res) = table_with_invalid_replicas
+
+        table_name = res["spec"]["tableName"]
+
+        # Check DynamoDB Table exists
+        assert self.table_exists(table_name)
+
+        # Wait for the terminal condition to be set
+        max_wait_seconds = 120
+        interval_seconds = 10
+        start_time = time.time()
+        terminal_condition_set = False
+
+        while time.time() - start_time < max_wait_seconds:
+            cr = k8s.get_resource(ref)
+            if cr and "status" in cr and "conditions" in cr["status"]:
+                for condition_obj in cr["status"]["conditions"]:
+                    if condition_obj["type"] == "ACK.Terminal" and condition_obj["status"] == "True":
+                        terminal_condition_set = True
+                        # Verify the error message
+                        assert "table must have DynamoDB Streams enabled with StreamViewType set to NEW_AND_OLD_IMAGES" in condition_obj[
+                            "message"]
+                        break
+
+            if terminal_condition_set:
+                break
+
+            time.sleep(interval_seconds)
+
+        assert terminal_condition_set, "Terminal condition was not set for invalid StreamSpecification"
+
+    def test_staged_replicas_and_gsi_updates(self):
+        table_name = random_suffix_name("table-replicas-gsi", 32)
+
+        # Step 1: Create table with single GSI and single replica using the template
+        replacements = REPLACEMENT_VALUES.copy()
+        replacements["TABLE_NAME"] = table_name
+        replacements["REPLICA_REGION_1"] = REPLICA_REGION_1
+
+        resource_data = load_dynamodb_resource(
+            "table_with_gsi_and_replicas",
+            additional_replacements=replacements,
+        )
+
+        # Create the k8s resource
+        ref = k8s.CustomResourceReference(
+            CRD_GROUP, CRD_VERSION, RESOURCE_PLURAL,
+            table_name, namespace="default",
+        )
```

Comment:
if this test fail, L528 will never be executed which will leave a dangling DDB Table in our accounts. Ideally you want to use a function that provisions the table, yields the info you need, and calls a delete function at the end.